lib/rubocop/cop/performance/caller.rb

```diff
@@ -8,32 +8,56 @@ module Performance
       #
       # @example
       #   # bad
-      #   caller[n]
+      #   caller[1]
       #   caller.first
       #
       #   # good
-      #   caller(n..n).first
+      #   caller(2..2).first
       #   caller(1..1).first
       class Caller < Cop
-        MSG = 'Use `caller(n..n)` instead of `caller[n]`.'.freeze
-        SCOPE_METHODS = %i[first []].freeze
+        MSG_BRACE =
+          'Use `caller(%<n>d..%<n>d).first` instead of `caller[%<m>d]`.'.freeze
+        MSG_FIRST =
+          'Use `caller(%<n>d..%<n>d).first` instead of `caller.first`.'.freeze
+
+        def_node_matcher :slow_caller?, <<-PATTERN
+          {
+            (send nil :caller)
+            (send nil :caller int)
+          }
+        PATTERN
 
         def_node_matcher :caller_with_scope_method?, <<-PATTERN
-          (send (send nil :caller ...) ${#{SCOPE_METHODS.map(&:inspect).join(' ')}} ...)
+          {
+            (send $_recv :first)
+            (send $_recv :[] int)
+          }
         PATTERN
 
         def on_send(node)
-          return unless caller_with_scope_method?(node) && slow_caller?(node)
+          recv = caller_with_scope_method?(node)
+          return unless slow_caller?(recv)
+
           add_offense(node, :selector)
```

Comment:
How about we change the highlight to `:expression` (default.) I think `:selector` becomes a bit weird with `#first`:

```ruby
caller.first
       ^^^^^
```

especially when the suggestion also contains `#first`. 🙂